llmebench/models/HuggingFace.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import json
+import requests
+from llmebench.models.model_base import ModelBase
+import os
+
+from tenacity import (
+    retry,
+    retry_if_exception_type,
+    stop_after_attempt,
+    wait_random_exponential,
+)
+
+API_TOKEN = os.environ["HUGGINGFACE_API_TOKEN"]
+
+
+def log_retry(retry_state):
+    if retry_state.attempt_number == 1:
+        return
+    logging.warning(
+        f"Request failed, retry attempt {retry_state.attempt_number}...")
+
+
+class HuggingFace(ModelBase):
+    def __init__(self, inference_api_url, max_tries=5, retry_exceptions=()):
+        self.max_tries = max_tries
+
+        self.inference_api_url = inference_api_url
+        self.api_token = API_TOKEN
+
+        # Instantiate retrying mechanism
+        self.prompt = retry(
```

Comment:
This is something that the `ModelBase` does to override the default prompt with extra decorations, we should just call it's constructor to do it instead of re-implementing it here. Something like

```python
super(HuggingFace, self).__init__(retry_exceptions=(...), **kwargs)
```
should be fine; Also we should retry on some common failures like an HTTP timeout maybe?

Reply:
https://huggingface.co/docs/api-inference/faq also states we may get 503 errors, something else to retry on as well